### ISSUE TITLE
Firefox does not work in full screen mode in a steam session

### ISSUE THREAD

**Issue by Vladislavuz** [ValveSoftware/SteamOS#723]
#### Your system information
 
* Steam client version: 1599174997
* SteamOS version: 2.0:1:2.195
* Opted into Steam client beta?: [Yes/No] Yes
* Opted into SteamOS beta?: [Yes/No] Yes
* Have you checked for updates in Settings > System?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen.
Sometimes I need to use the Firefox browser without switching to the desktop session, but in this mode Firefox does not work in full screen mode. There are also problems with displaying the context menu. Please see screenshots.
![2020-09-06 08-31-50](https://user-images.githubusercontent.com/30505454/92323340-9f55e080-f040-11ea-8102-c50361cebd80.JPG)
https://yadi.sk/i/OK4fNQNbkzdzoA
#### Steps for reproducing this issue:

1. Start Firefox
2. Try opening the context menu, hovering over an item to display a tooltip, or opening browser settings

**Comment by vicnett** [comment]
I'm having the same problem on the Steam Deck 😞

**Comment by Rahabib** [comment]
Ditto

**Comment by witcheslive** [comment]
somehow, this is still a problem that hasn't been fixed even in the steam deck era, it's a real pain that firefox doesn't really work in game mode

**Comment by TTimo** [comment]
Chrome can be installed via the non-steam tab when running gamescope, does that work for you?

**Comment by traverseda** [comment]
No, chrome's ad-block API is crippled in manifest v3, which will soon be the only option. Additionally I use firefox sync primarily.

**Comment by IamTheCarl** [comment]
For me, it's not just about the features of Chrome VS the features of Firefox (although they are still part of the equation).

Google owns the most popular browser, search engine, email platform, and more. I feel uneasy with how much of the internet they control, and the fact that they make attempts to force new standards on everyone for their benefit is a massive red flag to me.

The Steam Deck is applauded for being such an open platform. Valve's choice to run Linux instead of Windows on it is a huge factor. We don't have to live in fear of Microsoft restricting 3rd party software vendors like they've been doing.

I feel that the internet as a whole is at risk of having a similar issue. If we want it to remain open, we need more than one browser platform out there.

**Comment by witcheslive** [comment]
No, Chrome is not an option because I do not want to use the advertising company's browser.

**Comment by traverseda** [comment]
Now this is an issue with gamescope, that any developer could in theory pick up? Is the affected component open source?

**Comment by IamTheCarl** [comment]
I'm interested. [Is this the version on the Deck?](https://github.com/Plagman/gamescope)
The [official one in Valve's repositories](https://github.com/ValveSoftware/steamos-compositor) looks unmaintained, and I'm doubting that's what's currently on the Deck.

**Comment by IamTheCarl** [comment]
I came across this clever hack to access desktop mode from within gaming mode and it works very well for me: https://www.reddit.com/r/SteamDeck/comments/uddyat/desktop_mode_within_gaming_mode/
It has proven to be an effective workaround for me.

I will now repost the content of that post here *just in case* the original gets deleted.
```
---
#!/bin/sh
unset LD_PRELOAD
exec startplasma-wayland --xwayland --x11-display $DISPLAY --no-lockscreen --width 1280 --height 800 -- plasma_session 
---

Save it, make executable and add as a non-Steam app. From now on, you can use Desktop Mode as a yet another Gaming Mode app, with Steam overlay, convenient access to Steam Input configuration, being able to have a game running simultaneously and so on :)

It has some quirks, but hey, it's just a quick hack ;)
```

**Comment by amaturehr** [comment]
> No, chrome's ad-block API is crippled in manifest v3, which will soon be the only option. Additionally I use firefox sync primarily. Unfortunately chrome/chromium isn't really an alternative for me.

I feel the same about this

**Comment by traverseda** [comment]
As an aside, I finally decided to test chrome and is experiences similar issues. They're not as severe as it doesn't make as much use of that kind of menu-ing, but there are still some pretty big issues.

**Comment by IamTheCarl** [comment]
I too have tried Chrome (Ungoogled Chromium to be exact) and have expirenced the same you did.
I found half the video streaming sights I tried with it didn't work. That is likely **not** a SteamOS specific issue.

**Comment by traverseda** [comment]
I believe it's a gamescope issues, I can replicate the issues I'm having on standard linux running a gamescope session. Since gamescope is part of steamos, it is sort of a steamos issue.

**Comment by andrew-hill** [comment]
@traverseda there's a reported issue about this in Plagman/gamescope#607, if you'd like to add anything specific from your testing that may help narrow down the bug and/or clarify that it's a gamescope issue rather than a firefox one, that may be helpful?

And yeah, I'd agree it's a SteamOS issue as much as gamescope, since it's bundled in and impacts an intended use-case.

**Comment by andrew-hill** [comment]
I've just done a bit of digging within Firefox bugzilla, and this _could_ actually be a Firefox issue. There's a similar bug report with Firefox in Wayland (gamescope uses Wayland while desktop mode uses Xorg, as far as I can see, so there's another difference here).

I don't personally have the know-how to test this theory further, but maybe someone knows how to test Firefox on SteamOS with Wayland but _without_ gamescope? If the tooltips/menus fail in such a test, it may actually be a Firefox/Wayland issue rather than SteamOS/gamescope?

**Comment by SuperJumpBros** [comment]
i am also having this issue. i want to move to my steam deck as a main PC, but this one issue is just glaring me in the face. i only want it fixed.

**Comment by Guuiil** [comment]
I don't know if this will fix every problem you have but i think for the full screen the probleme is that the ratio of the firefox window that opens isnt the same as fullscreen so when steamOS only display the window it is not fullscreen

For me binding F11 in steamOS to a button and pressing it fixed it (desktop still always runs in background so if you get it to be fullscreen on the desktop it should work well in gaming mode)

**Comment by Tech-How** [comment]
I'm having this issue as well. Firefox always displays at 1280x800 in game mode and cannot be adjusted.

**Comment by Jau-Al** [comment]
Not being able to use full-screen is a disaster for video content. Did anyone find something ?

**Comment by matte-schwartz** [comment]
basically just what @Guuiil found when I took a look recently, you need to trigger Firefox to go fullscreen on its own with F11 on a keyboard. maybe there's a way to script it so this is done automatically as they suggested, although I've not taken a look at how you'd go about this myself.

**Comment by Tech-How** [comment]
The closest I got was setting `full-screen-api.ignore-widgets` to true in about:config, but that still doesn't fix the 800p issue. It will allow you to fullscreen the browser from the upper right menu, at least. You can also use `--fullscreen` in the launch options if you're launching a PWA directly from the game UI.
